dpd/p4/constants.p4

```diff
@@ -1,36 +1,43 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/
+//
+// Copyright 2025 Oxide Computer Company
+
 const bit<16> L2_ISOLATED_FLAG = 0x8000;
-#define IS_SERVICE(p) ((p) == USER_SPACE_SERVICE_PORT)
-
-//TODO these all need to be bigger. Early experimentation is showing that this
-//is going to need to come either through ATCAM/ALPM or code restructuring.
-const int IPV4_NAT_TABLE_SIZE		= 1024; // nat routing table
-const int IPV6_NAT_TABLE_SIZE		= 1024; // nat routing table
-const int IPV4_LPM_SIZE			= 8192; // ipv4 forwarding table
-const int IPV6_LPM_SIZE			= 1024; // ipv6 forwarding table
-
-const int IPV4_ARP_SIZE			= 512;  // arp cache
-const int IPV6_NEIGHBOR_SIZE		= 512;  // ipv6 neighbor cache
-const int SWITCH_IPV4_ADDRS_SIZE	= 512;  // ipv4 addrs assigned to our ports
-const int SWITCH_IPV6_ADDRS_SIZE	= 512;  // ipv6 addrs assigned to our ports
-
-const bit<8> SC_FWD_FROM_USERSPACE	= 0x00;
-const bit<8> SC_FWD_TO_USERSPACE	= 0x01;
-const bit<8> SC_ICMP_NEEDED		= 0x02;
-const bit<8> SC_ARP_NEEDED		= 0x03;
-const bit<8> SC_NEIGHBOR_NEEDED		= 0x04;
-const bit<8> SC_INVALID			= 0xff;
+
+// TODO: these all need to be bigger. Early experimentation is showing that this
+// is going to need to come either through ATCAM/ALPM or code restructuring.
+const int IPV4_NAT_TABLE_SIZE       = 1024; // nat routing table
+const int IPV6_NAT_TABLE_SIZE       = 1024; // nat routing table
+const int IPV4_LPM_SIZE             = 8192; // ipv4 forwarding table
+const int IPV6_LPM_SIZE             = 1024; // ipv6 forwarding table
+const int IPV4_ARP_SIZE             = 512;  // arp cache
+const int IPV6_NEIGHBOR_SIZE        = 512;  // ipv6 neighbor cache
+const int SWITCH_IPV4_ADDRS_SIZE    = 512;  // ipv4 addrs assigned to our ports
+const int SWITCH_IPV6_ADDRS_SIZE    = 512;  // ipv6 addrs assigned to our ports
+const int IPV4_MULTICAST_TABLE_SIZE = 1024; // multicast routing table(s) for IPv4
```

Comment:
How did you arrive at a size of 1024 for all of the multicast tables?  I suspect that when we add customer support for IPv6, we're going to have to bump up the existing IPv6 tables, and we don't have a lot of headroom.  Unless we know that we need the multicast tables to be this big, I would be inclined to start with something much smaller and grow them if the customer hits the limit.  That will be much lower risk than trying to shrink them after this is in use at a customer site.

Reply:
@Nieuwejaar I thought 1024 was starting small, based on what mcast-interested customers had been bringing up (which was larger). I could break up the V6 constants as they're used for different table components, but maybe @rcgoodfellow has thoughts here.